java-util/src/main/java/io/druid/java/util/common/CompressionUtils.java

```diff
@@ -263,7 +265,9 @@ public InputStream openStream() throws IOException
       ZipEntry entry;
       while ((entry = zipIn.getNextEntry()) != null) {
         final File file = new File(outDir, entry.getName());
-        Files.asByteSink(file).writeFrom(zipIn);
+
+        NativeIO.chunkedCopy(zipIn, file);
```

Comment:
If I recall correctly the zip implementation might actually do Memory Mapping under the hood in Liniux JVMs. If that is the case, would fadvise even work?

Reply:
I guess it could on the output file, and the input file would be dropped after a download cleaned up anyways.

Reply:
I need to have a look at that. But both would enter the os cache. Just dropping it from the cache does not solve the issue, as the file that used to be in that cache needs to be reloade. You just never want it to enter the cache.

Reply:
So yes, zip can use mmap, but only does this for the original zip file.

`private static native long open(String name, int mode, long lastModified,
                                    boolean usemmap) throws IOException;`